### ISSUE TITLE
Crash on build

### ISSUE THREAD

**Issue by kcoderhtml** [hashicorp/packer#12283]
#### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

When filing a bug, please include the following headings if possible. Any
example text in this template can be deleted.

When I build my packer file I get a crash

#### Reproduction Steps

Run packer build.

### Packer version

From `packer version` Packer v1.8.6

### Simplified Packer Template

[Here](https://github.com/ChristianLempa/boilerplates/tree/main/packer/proxmox/ubuntu-server-jammy-docker)

### Operating system and Environment details

Amd 64, Pop OS

### Log Fragments and crash.log files
```
2023/02/20 13:26:12 [INFO] Packer version: 1.8.6 [go1.18.9 linux amd64]
2023/02/20 13:26:12 [TRACE] discovering plugins in /usr/bin
2023/02/20 13:26:12 [TRACE] discovering plugins in /home/kierank/.config/packer/plugins
2023/02/20 13:26:12 [TRACE] discovering plugins in .
2023/02/20 13:26:12 [INFO] PACKER_CONFIG env var not set; checking the default config file path
2023/02/20 13:26:12 [INFO] PACKER_CONFIG env var set; attempting to open config file: /home/kierank/.packerconfig
2023/02/20 13:26:12 [WARN] Config file doesn't exist: /home/kierank/.packerconfig
2023/02/20 13:26:12 [INFO] Setting cache directory: /home/kierank/.cache/packer
2023/02/20 13:26:12 [TRACE] validateValue: not active for proxmox_api_token_id, so skipping
2023/02/20 13:26:12 [TRACE] validateValue: not active for proxmox_api_token_secret, so skipping
2023/02/20 13:26:12 [TRACE] validateValue: not active for proxmox_api_url, so skipping
2023/02/20 13:26:12 [TRACE] Starting internal plugin packer-builder-proxmox
2023/02/20 13:26:12 Starting plugin: /usr/bin/packer []string{"/usr/bin/packer", "plugin", "packer-builder-proxmox"}
2023/02/20 13:26:12 Waiting for RPC address for: /usr/bin/packer
2023/02/20 13:26:12 packer-builder-proxmox plugin: [INFO] Packer version: 1.8.6 [go1.18.9 linux amd64]
2023/02/20 13:26:12 packer-builder-proxmox plugin: [INFO] PACKER_CONFIG env var not set; checking the default config file path
2023/02/20 13:26:12 packer-builder-proxmox plugin: [INFO] PACKER_CONFIG env var set; attempting to open config file: /home/kierank/.packerconfig
2023/02/20 13:26:12 packer-builder-proxmox plugin: [WARN] Config file doesn't exist: /home/kierank/.packerconfig
2023/02/20 13:26:12 packer-builder-proxmox plugin: [INFO] Setting cache directory: /home/kierank/.cache/packer
2023/02/20 13:26:12 packer-builder-proxmox plugin: args: []string{"packer-builder-proxmox"}
2023/02/20 13:26:12 packer-builder-proxmox plugin: Plugin address: unix /tmp/packer-plugin3518785845
2023/02/20 13:26:12 packer-builder-proxmox plugin: Waiting for connection...
2023/02/20 13:26:12 Received unix RPC address for /usr/bin/packer: addr is /tmp/packer-plugin3518785845
2023/02/20 13:26:12 packer-builder-proxmox plugin: Serving a plugin connection...
2023/02/20 13:26:12 packer-builder-proxmox plugin: Number of sockets 0 is too small, using default: 1
2023/02/20 13:26:12 packer-builder-proxmox plugin: CPU type not set, using default 'kvm64'
2023/02/20 13:26:12 packer-builder-proxmox plugin: OS not set, using default 'other'
2023/02/20 13:26:12 packer-builder-proxmox plugin: Disk 0 cache mode not set, using default 'none'
2023/02/20 13:26:12 [TRACE] Starting internal plugin packer-provisioner-shell
2023/02/20 13:26:12 Starting plugin: /usr/bin/packer []string{"/usr/bin/packer", "plugin", "packer-provisioner-shell"}
2023/02/20 13:26:12 Waiting for RPC address for: /usr/bin/packer
2023/02/20 13:26:12 packer-provisioner-shell plugin: [INFO] Packer version: 1.8.6 [go1.18.9 linux amd64]
2023/02/20 13:26:12 packer-provisioner-shell plugin: [INFO] PACKER_CONFIG env var not set; checking the default config file path
2023/02/20 13:26:12 packer-provisioner-shell plugin: [INFO] PACKER_CONFIG env var set; attempting to open config file: /home/kierank/.packerconfig
2023/02/20 13:26:12 packer-provisioner-shell plugin: [WARN] Config file doesn't exist: /home/kierank/.packerconfig
2023/02/20 13:26:12 packer-provisioner-shell plugin: [INFO] Setting cache directory: /home/kierank/.cache/packer
2023/02/20 13:26:12 packer-provisioner-shell plugin: args: []string{"packer-provisioner-shell"}
2023/02/20 13:26:12 packer-provisioner-shell plugin: Plugin address: unix /tmp/packer-plugin3962937989
2023/02/20 13:26:12 packer-provisioner-shell plugin: Waiting for connection...
2023/02/20 13:26:12 Received unix RPC address for /usr/bin/packer: addr is /tmp/packer-plugin3962937989
2023/02/20 13:26:12 packer-provisioner-shell plugin: Serving a plugin connection...
2023/02/20 13:26:12 [TRACE] Starting internal plugin packer-provisioner-file
2023/02/20 13:26:12 Starting plugin: /usr/bin/packer []string{"/usr/bin/packer", "plugin", "packer-provisioner-file"}
2023/02/20 13:26:12 Waiting for RPC address for: /usr/bin/packer
2023/02/20 13:26:12 packer-provisioner-file plugin: [INFO] Packer version: 1.8.6 [go1.18.9 linux amd64]
2023/02/20 13:26:12 packer-provisioner-file plugin: [INFO] PACKER_CONFIG env var not set; checking the default config file path
2023/02/20 13:26:12 packer-provisioner-file plugin: [INFO] PACKER_CONFIG env var set; attempting to open config file: /home/kierank/.packerconfig
2023/02/20 13:26:12 packer-provisioner-file plugin: [WARN] Config file doesn't exist: /home/kierank/.packerconfig
2023/02/20 13:26:12 packer-provisioner-file plugin: [INFO] Setting cache directory: /home/kierank/.cache/packer
2023/02/20 13:26:12 packer-provisioner-file plugin: args: []string{"packer-provisioner-file"}
2023/02/20 13:26:12 packer-provisioner-file plugin: Plugin address: unix /tmp/packer-plugin1396439442
2023/02/20 13:26:12 packer-provisioner-file plugin: Waiting for connection...
2023/02/20 13:26:12 Received unix RPC address for /usr/bin/packer: addr is /tmp/packer-plugin1396439442
2023/02/20 13:26:12 packer-provisioner-file plugin: Serving a plugin connection...
2023/02/20 13:26:12 [TRACE] Starting internal plugin packer-provisioner-shell
2023/02/20 13:26:12 Starting plugin: /usr/bin/packer []string{"/usr/bin/packer", "plugin", "packer-provisioner-shell"}
2023/02/20 13:26:12 Waiting for RPC address for: /usr/bin/packer
2023/02/20 13:26:12 packer-provisioner-shell plugin: [INFO] Packer version: 1.8.6 [go1.18.9 linux amd64]
2023/02/20 13:26:12 packer-provisioner-shell plugin: [INFO] PACKER_CONFIG env var not set; checking the default config file path
2023/02/20 13:26:12 packer-provisioner-shell plugin: [INFO] PACKER_CONFIG env var set; attempting to open config file: /home/kierank/.packerconfig
2023/02/20 13:26:12 packer-provisioner-shell plugin: [WARN] Config file doesn't exist: /home/kierank/.packerconfig
2023/02/20 13:26:12 packer-provisioner-shell plugin: [INFO] Setting cache directory: /home/kierank/.cache/packer
2023/02/20 13:26:12 packer-provisioner-shell plugin: args: []string{"packer-provisioner-shell"}
2023/02/20 13:26:12 packer-provisioner-shell plugin: Plugin address: unix /tmp/packer-plugin4157463251
2023/02/20 13:26:12 packer-provisioner-shell plugin: Waiting for connection...
2023/02/20 13:26:12 Received unix RPC address for /usr/bin/packer: addr is /tmp/packer-plugin4157463251
2023/02/20 13:26:12 packer-provisioner-shell plugin: Serving a plugin connection...
2023/02/20 13:26:12 [TRACE] Starting internal plugin packer-provisioner-shell
2023/02/20 13:26:12 Starting plugin: /usr/bin/packer []string{"/usr/bin/packer", "plugin", "packer-provisioner-shell"}
2023/02/20 13:26:12 Waiting for RPC address for: /usr/bin/packer
2023/02/20 13:26:12 packer-provisioner-shell plugin: [INFO] Packer version: 1.8.6 [go1.18.9 linux amd64]
2023/02/20 13:26:12 packer-provisioner-shell plugin: [INFO] PACKER_CONFIG env var not set; checking the default config file path
2023/02/20 13:26:12 packer-provisioner-shell plugin: [INFO] PACKER_CONFIG env var set; attempting to open config file: /home/kierank/.packerconfig
2023/02/20 13:26:12 packer-provisioner-shell plugin: [WARN] Config file doesn't exist: /home/kierank/.packerconfig
2023/02/20 13:26:12 packer-provisioner-shell plugin: [INFO] Setting cache directory: /home/kierank/.cache/packer
2023/02/20 13:26:12 packer-provisioner-shell plugin: args: []string{"packer-provisioner-shell"}
2023/02/20 13:26:12 packer-provisioner-shell plugin: Plugin address: unix /tmp/packer-plugin514328065
2023/02/20 13:26:12 packer-provisioner-shell plugin: Waiting for connection...
2023/02/20 13:26:12 Received unix RPC address for /usr/bin/packer: addr is /tmp/packer-plugin514328065
2023/02/20 13:26:12 packer-provisioner-shell plugin: Serving a plugin connection...
2023/02/20 13:26:12 ui: ubuntu-server-jammy.proxmox.ubuntu-server-jammy: output will be in this color.
2023/02/20 13:26:12 ui:
2023/02/20 13:26:12 Build debug mode: false
2023/02/20 13:26:12 Force build: false
2023/02/20 13:26:12 On error:
2023/02/20 13:26:12 Waiting on builds to complete...
2023/02/20 13:26:12 Starting build run: ubuntu-server-jammy.proxmox.ubuntu-server-jammy
2023/02/20 13:26:12 Running builder:
2023/02/20 13:26:12 [INFO] (telemetry) Starting builder proxmox.ubuntu-server-jammy
2023/02/20 13:26:12 packer-builder-proxmox plugin: using token auth
2023/02/20 13:26:12 packer-builder-proxmox plugin: No URLs were provided to Step Download. Continuing...
2023/02/20 13:26:12 ui: ==> ubuntu-server-jammy.proxmox.ubuntu-server-jammy: Creating VM
2023/02/20 13:26:12 packer-builder-proxmox plugin: panic: runtime error: invalid memory address or nil pointer dereference
2023/02/20 13:26:12 packer-builder-proxmox plugin: [signal SIGSEGV: segmentation violation code=0x1 addr=0x40 pc=0x2eedf37]
2023/02/20 13:26:12 packer-builder-proxmox plugin:
2023/02/20 13:26:12 packer-builder-proxmox plugin: goroutine 161 [running]:
2023/02/20 13:26:12 packer-builder-proxmox plugin: github.com/Telmate/proxmox-api-go/proxmox.(*Client).CreateQemuVm(0xc0006c4140, {0xc000a760d8, 0x7}, 0x51e4570?)
2023/02/20 13:26:12 packer-builder-proxmox plugin: 	github.com/Telmate/proxmox-api-go@v0.0.0-20221209160129-bb95d9a81e7a/proxmox/client.go:564 +0x297
2023/02/20 13:26:12 packer-builder-proxmox plugin: github.com/Telmate/proxmox-api-go/proxmox.ConfigQemu.CreateVm({0x0, {0xc0004c7230, 0x13}, {0x51e4570, 0x19}, {0x0, 0x0}, {0x0, 0x0}, 0xc0005a2000, ...}, ...)
2023/02/20 13:26:12 packer-builder-proxmox plugin: 	github.com/Telmate/proxmox-api-go@v0.0.0-20221209160129-bb95d9a81e7a/proxmox/config_qemu.go:213 +0x12af
2023/02/20 13:26:12 packer-builder-proxmox plugin: github.com/hashicorp/packer-plugin-proxmox/builder/proxmox/iso.(*isoVMCreator).Create(_, _, {0x0, {0xc0004c7230, 0x13}, {0x51e4570, 0x19}, {0x0, 0x0}, {0x0, ...}, ...}, ...)
2023/02/20 13:26:12 packer-builder-proxmox plugin: 	github.com/hashicorp/packer-plugin-proxmox@v1.1.1/builder/proxmox/iso/builder.go:64 +0xe5
2023/02/20 13:26:12 packer-builder-proxmox plugin: github.com/hashicorp/packer-plugin-proxmox/builder/proxmox/common.(*stepStartVM).Run(0xc0006bc0b0, {0x5196368?, 0xc0006ca080?}, {0x59e2f28, 0xc0006b8120})
2023/02/20 13:26:12 packer-builder-proxmox plugin: 	github.com/hashicorp/packer-plugin-proxmox@v1.1.1/builder/proxmox/common/step_start_vm.go:94 +0x9ac
2023/02/20 13:26:12 packer-builder-proxmox plugin: github.com/hashicorp/packer-plugin-sdk/multistep.(*BasicRunner).Run(0xc0006b81e0, {0x59e2128, 0xc0006ca080}, {0x59e2f28, 0xc0006b8120})
2023/02/20 13:26:12 packer-builder-proxmox plugin: 	github.com/hashicorp/packer-plugin-sdk@v0.3.4/multistep/basic_runner.go:73 +0x2f3
2023/02/20 13:26:12 packer-builder-proxmox plugin: github.com/hashicorp/packer-plugin-proxmox/builder/proxmox/common.(*Builder).Run(0xc0006d6000, {0x59e2128, 0xc0006ca080}, {0x59e6fd0?, 0xc0006b80f0}, {0x59a5f20?, 0xc0006b0020}, {0x59e2f28?, 0xc0006b8120?})
2023/02/20 13:26:12 packer-builder-proxmox plugin: 	github.com/hashicorp/packer-plugin-proxmox@v1.1.1/builder/proxmox/common/builder.go:100 +0xde3
2023/02/20 13:26:12 packer-builder-proxmox plugin: github.com/hashicorp/packer-plugin-proxmox/builder/proxmox/iso.(*Builder).Run(0xc000b2fc00, {0x59e2128, 0xc0006ca080}, {0x59e6fd0, 0xc0006b80f0}, {0x59a5f20, 0xc0006b0020})
2023/02/20 13:26:12 packer-builder-proxmox plugin: 	github.com/hashicorp/packer-plugin-proxmox@v1.1.1/builder/proxmox/iso/builder.go:54 +0x3c5
2023/02/20 13:26:12 packer-builder-proxmox plugin: github.com/hashicorp/packer-plugin-sdk/rpc.(*BuilderServer).Run(0xc000a93680, 0xaa2030?, 0xc000aa2010)
2023/02/20 13:26:12 packer-builder-proxmox plugin: 	github.com/hashicorp/packer-plugin-sdk@v0.3.4/rpc/builder.go:120 +0x1e3
2023/02/20 13:26:12 packer-builder-proxmox plugin: reflect.Value.call({0xc000566c00?, 0xc000532288?, 0x13?}, {0x518b294, 0x4}, {0xc000b72ef8, 0x3, 0x3?})
2023/02/20 13:26:12 packer-builder-proxmox plugin: 	reflect/value.go:556 +0x845
2023/02/20 13:26:12 packer-builder-proxmox plugin: reflect.Value.Call({0xc000566c00?, 0xc000532288?, 0x3?}, {0xc00112d6f8, 0x3, 0x3})
2023/02/20 13:26:12 packer-builder-proxmox plugin: 	reflect/value.go:339 +0xbf
2023/02/20 13:26:12 packer-builder-proxmox plugin: net/rpc.(*service).call(0xc000a936c0, 0xc000a51747?, 0x0?, 0xc000b0a500, 0xc000b6a300, 0x7c2765?, {0x40aac20?, 0xc000aa200c?, 0x7c26e6?}, {0x3f7b3e0, ...}, ...)
2023/02/20 13:26:12 packer-builder-proxmox plugin: 	net/rpc/server.go:381 +0x226
2023/02/20 13:26:12 packer-builder-proxmox plugin: created by net/rpc.(*Server).ServeCodec
2023/02/20 13:26:12 packer-builder-proxmox plugin: 	net/rpc/server.go:478 +0x3fe
2023/02/20 13:26:12 [INFO] (telemetry) ending proxmox.ubuntu-server-jammy
2023/02/20 13:26:12 ui error: Build 'ubuntu-server-jammy.proxmox.ubuntu-server-jammy' errored after 16 milliseconds 659 microseconds: unexpected EOF
2023/02/20 13:26:12 /usr/bin/packer: plugin process exited
2023/02/20 13:26:12 ui:
==> Wait completed after 16 milliseconds 686 microseconds
2023/02/20 13:26:12 machine readable: error-count []string{"1"}
2023/02/20 13:26:12 ui error:
==> Some builds didn't complete successfully and had errors:
2023/02/20 13:26:12 machine readable: ubuntu-server-jammy.proxmox.ubuntu-server-jammy,error []string{"unexpected EOF"}
2023/02/20 13:26:12 ui error: --> ubuntu-server-jammy.proxmox.ubuntu-server-jammy: unexpected EOF
2023/02/20 13:26:12 ui:
==> Builds finished but no artifacts were created.
2023/02/20 13:26:12 [INFO] (telemetry) Finalizing.
2023/02/20 13:26:12 waiting for all plugin processes to complete...
2023/02/20 13:26:12 /usr/bin/packer: plugin process exited
2023/02/20 13:26:12 /usr/bin/packer: plugin process exited
2023/02/20 13:26:12 /usr/bin/packer: plugin process exited
2023/02/20 13:26:12 /usr/bin/packer: plugin process exited
```

**Comment by lbajolet-hashicorp** [comment]
Hi @kcoderhtml,

The issue you're reporting is already documented on the Proxmox plugin, and from what I investigated, is due to the upstream Proxmox Go API.

Unfortunately, this is not something we can easily fix or reproduce, and since this is out-of-scope for Packer and its plugins, not something we can fix right now.

I'll transfer your issue over there since this is related to the Proxmox plugin rather than Packer itself.

See for example issue: https://github.com/hashicorp/packer-plugin-proxmox/issues/137

Thanks for reporting this
